umm-spec-lib/src/cmr/umm_spec/field_update.clj

```diff
@@ -167,38 +179,55 @@
   ;; Any information other than the home page url value under :ContactInformation
   ;; in update-value is not used.
   [update-type umm update-field update-value find-value]
-  (if (seq (get-in umm update-field))
-    ;; For each entry in update-field, if we find it using the find params,
-    ;; update only the fields supplied in update-value with nils removed, except for the
-    ;; ContactInformation part of the update-value.
-    (update-in umm update-field #(distinct (map (fn [x]
-                                                  (if (value-matches? find-value x)
-                                                    (data-center-update x update-value)
-                                                    x))
-                                                %)))
-    umm))
+  ;; For each entry in update-field, if we find it using the find params,
+  ;; update only the fields supplied in update-value with nils removed, except for the
+  ;; ContactInformation part of the update-value.
+  (update-in umm update-field #(distinct (map (fn [x]
+                                                (if (value-matches? find-value x)
+                                                  (data-center-update x update-value)
+                                                  x))
+                                              %))))
+
+
+(defn- apply-list-update-nested
+  "Apply the umm list update when the update-field is a supported nested field."
+  [update-type umm update-field update-value find-value partial-update-fn]
+  (if (re-matches #":find-and.*" (str update-type))
```

Comment:
We should have a var that contains a collection of the update-types that support a find value rather than using a regex. Then check if that update-type is in the list. Same for the other function below.

Reply:
Thanks. I thought about that too. Done.